packages/generator/src/generator.js

```diff
@@ -23,6 +24,12 @@ export default class Generator {
       isUrl(this.options.build.publicPath) ? '' : this.options.build.publicPath
     )
     this.generatedRoutes = new Set()
+
+    // Shared payload
+    this._payload = null
+    this.setPayload = (payload) => {
```

Comment:
Would `getter/setter` for `payload` make more sense here ?

Reply:
We may need something like a proxy then for deep watch. setPayload can ensure it is deep assigned and is simple :)

Reply:
This make me a little confuse, we have `this.setPayload` which is never called.

We also have another `setPayload` as a [scoped var](https://github.com/nuxt/nuxt.js/pull/7422/files#diff-22690cd173b27a10986baff5efcd4ac4R280)

Reply:
The one set in the constructor is used by `export:before` hook to set shared payload.

The one in scoped var is used for `export:route` hook.

Reply:
Ah, I didn’t realize it’s passed by this.

Reply:
I think I might have understood what’s happening here, so we have a global payload, each route has a scoped payload, what we’re doing is to make sure default value is always populated to updated payload.